smmregrid/gridinspector.py

```diff
@@ -0,0 +1,132 @@
+"""GridInspector class module"""
+
+import xarray as xr
+from smmregrid.log import setup_logger
+from .gridtype import GridType
+
+
+class GridInspector():
+    """Class to investigate data and detect its GridType() object"""
+
+    def __init__(self, data, cdo_weights=False, extra_dims=None,
+                 clean=True, loglevel='warning'):
+        """
+        GridInspector class to detect information on the data, based on GridType class
+
+        Parameters:
+            data (xr.Datase or xr.DataArray): The input dataset.
+            clean (bool): apply the cleaning of grids which are assumed to be not relevant 
+                          for regridding purposes (e.g. bounds)
+            cdo_weights (bool): if the data provided are cdo weights instead of data to be regridded
+            extra_dims(dict): Extra dimensions to be added and passed to GridType
+            loglevel: The loglevel that you want you use
+        """
+        self.loggy = setup_logger(name='smmregrid.GridInspect', level=loglevel)
+        self.loglevel = loglevel
+        self.extra_dims = extra_dims
+        self.data = data
+        self.cdo_weights = cdo_weights
```

Comment:
why is `self.cdo_weights` needed? Would it not be enough to use `if self.weights` ?

Reply:
well `cdo_weigths` is a boolean for checks, `weigths` is the real weigths value...